Makefile

```diff
@@ -21,7 +21,10 @@ RUBY_ENV += RUBY_GC_OLDMALLOC_LIMIT=40125988
 RUBY_ENV += RUBY_GC_OLDMALLOC_LIMIT_MAX=72226778
 RUBY_ENV += RUBY_GC_OLDMALLOC_LIMIT_GROWTH_FACTOR=1.2
 
-
+container_runtime ?= $(shell docker --version | grep -q podman && echo podman || echo docker)
+oracle_podman_flags ?= --mount=type=bind,source=$(ORACLE_DATA_DIR)/oracle-database,destination=/mnt,relabel=shared
```

Comment:
I use podman with Fedora 34 and it works like a charm without using that (no customization so far except docker-podman package which is a wrapper around podman)

What is the error you get?

Reply:
I'm also running Fedora 34. When I use the proposed start-up command, the database starts up. Otherwise see log below. btw I assume then relabel can be `z` or `Z`. `shared` is perhaps not necessary.

```
$ sudo mkdir -p /var/lib/oracleX/oracle-database
$ sudo chown user:user /var/lib/oracleX /var/lib/oracleX/oracle-database
$ sudo chmod 770 /var/lib/oracleX /var/lib/oracleX/oracle-database

$ ORACLE_DATA_DIR=/var/lib/oracleX make oracle-database 
[ "" = "true" ] || docker start oracle-database || docker run \
	--shm-size=6gb \
	-p 1521:1521 -p 5500:5500 \
	--name oracle-database \
	-v /var/lib/oracleX/oracle-database:/opt/oracle/oradata \
	-e ORACLE_PDB=systempdb \
	-e ORACLE_SID=threescale \
	-e ORACLE_PWD=threescalepass \
	-e ORACLE_CHARACTERSET=AL32UTF8 \
	quay.io/3scale/oracle:19.3.0-ee-ci-prebuilt
Error: no container with name or ID "oracle-database" found: no such container
WARN[0000] additional gid=54323 is not present in the user namespace, skip setting it 
WARN[0000] additional gid=54324 is not present in the user namespace, skip setting it 
WARN[0000] additional gid=54325 is not present in the user namespace, skip setting it 
WARN[0000] additional gid=54326 is not present in the user namespace, skip setting it 
WARN[0000] additional gid=54330 is not present in the user namespace, skip setting it 
cat: /sys/fs/cgroup/memory/memory.limit_in_bytes: No such file or directory
cat: /sys/fs/cgroup/memory/memory.limit_in_bytes: No such file or directory
/opt/oracle/runOracle.sh: line 102: [: -lt: unary operator expected
ORACLE PASSWORD FOR SYS, SYSTEM AND PDBADMIN: threescalepass

LSNRCTL for Linux: Version 19.0.0.0.0 - Production on 02-AUG-2021 06:20:39

Copyright (c) 1991, 2019, Oracle.  All rights reserved.

Starting /opt/oracle/product/19c/dbhome_1/bin/tnslsnr: please wait...

TNSLSNR for Linux: Version 19.0.0.0.0 - Production
System parameter file is /opt/oracle/product/19c/dbhome_1/network/admin/listener.ora
Log messages written to /opt/oracle/diag/tnslsnr/7f93b0df6057/listener/alert/log.xml
Listening on: (DESCRIPTION=(ADDRESS=(PROTOCOL=ipc)(KEY=EXTPROC1)))
Listening on: (DESCRIPTION=(ADDRESS=(PROTOCOL=tcp)(HOST=0.0.0.0)(PORT=1521)))

Connecting to (DESCRIPTION=(ADDRESS=(PROTOCOL=IPC)(KEY=EXTPROC1)))
STATUS of the LISTENER
------------------------
Alias                     LISTENER
Version                   TNSLSNR for Linux: Version 19.0.0.0.0 - Production
Start Date                02-AUG-2021 06:20:39
Uptime                    0 days 0 hr. 0 min. 0 sec
Trace Level               off
Security                  ON: Local OS Authentication
SNMP                      OFF
Listener Parameter File   /opt/oracle/product/19c/dbhome_1/network/admin/listener.ora
Listener Log File         /opt/oracle/diag/tnslsnr/7f93b0df6057/listener/alert/log.xml
Listening Endpoints Summary...
  (DESCRIPTION=(ADDRESS=(PROTOCOL=ipc)(KEY=EXTPROC1)))
  (DESCRIPTION=(ADDRESS=(PROTOCOL=tcp)(HOST=0.0.0.0)(PORT=1521)))
The listener supports no services
The command completed successfully
[WARNING] [DBT-10328] Specified GDB Name (THREESCALE) may have a potential conflict with an already existing database on the system.
   ACTION: Specify a different GDB Name that does not conflict with existing databases on the system.
[FATAL] [DBT-10317] Specified SID Name (THREESCALE) already exists.
   ACTION: Specify a different SID Name that does not already exist.
[ 2020-05-20 22:44:05.737 UTC ] Prepare for db operation
DBCA_PROGRESS : 8%
[ 2020-05-20 22:44:05.810 UTC ] Copying database files
DBCA_PROGRESS : 31%
[ 2020-05-20 22:45:44.076 UTC ] Creating and starting Oracle instance
DBCA_PROGRESS : 32%
DBCA_PROGRESS : 36%
DBCA_PROGRESS : 40%
DBCA_PROGRESS : 43%
DBCA_PROGRESS : 46%
[ 2020-05-20 22:56:40.781 UTC ] Completing Database Creation
DBCA_PROGRESS : 51%
DBCA_PROGRESS : 54%
[ 2020-05-20 23:03:54.851 UTC ] Creating Pluggable Databases
DBCA_PROGRESS : 58%
DBCA_PROGRESS : 77%
[ 2020-05-20 23:04:02.706 UTC ] Executing Post Configuration Actions
DBCA_PROGRESS : 100%
[ 2020-05-20 23:04:02.707 UTC ] Database creation complete. For details check the logfiles at:
 /opt/oracle/cfgtoollogs/dbca/THREESCALE.
Database Information:
Global Database Name:THREESCALE
System Identifier(SID):THREESCALE

SQL*Plus: Release 19.0.0.0.0 - Production on Mon Aug 2 06:20:42 2021
Version 19.3.0.0.0

Copyright (c) 1982, 2019, Oracle.  All rights reserved.

Connected to an idle instance.

SQL>    ALTER SYSTEM SET control_files='/opt/oracle/oradata/THREESCALE/control01.ctl' scope=spfile
*
ERROR at line 1:
ORA-01034: ORACLE not available
Process ID: 0
Session ID: 0 Serial number: 0


SQL>    ALTER SYSTEM SET local_listener=''
*
ERROR at line 1:
ORA-01034: ORACLE not available
Process ID: 0
Session ID: 0 Serial number: 0


SQL>    ALTER PLUGGABLE DATABASE SYSTEMPDB SAVE STATE
*
ERROR at line 1:
ORA-01034: ORACLE not available
Process ID: 0
Session ID: 0 Serial number: 0


SQL> BEGIN DBMS_XDB_CONFIG.SETGLOBALPORTENABLED (TRUE); END;

*
ERROR at line 1:
ORA-01034: ORACLE not available
Process ID: 0
Session ID: 0 Serial number: 0


SQL> Disconnected
mkdir: cannot create directory '/opt/oracle/oradata': Permission denied
mv: failed to access '/opt/oracle/oradata/dbconfig/THREESCALE/': Permission denied
mv: failed to access '/opt/oracle/oradata/dbconfig/THREESCALE/': Permission denied
mv: failed to access '/opt/oracle/oradata/dbconfig/THREESCALE/': Permission denied
mv: failed to access '/opt/oracle/oradata/dbconfig/THREESCALE/': Permission denied
mv: failed to access '/opt/oracle/oradata/dbconfig/THREESCALE/': Permission denied
cp: failed to access '/opt/oracle/oradata/dbconfig/THREESCALE/': Permission denied
ln: failed to create symbolic link '/opt/oracle/product/19c/dbhome_1/network/admin/sqlnet.ora': File exists
ln: failed to create symbolic link '/opt/oracle/product/19c/dbhome_1/network/admin/listener.ora': File exists
ln: failed to create symbolic link '/opt/oracle/product/19c/dbhome_1/network/admin/tnsnames.ora': File exists
cp: cannot stat '/opt/oracle/oradata/dbconfig/THREESCALE/oratab': Permission denied
The Oracle base remains unchanged with value /opt/oracle
#####################################
########### E R R O R ###############
DATABASE SETUP WAS NOT SUCCESSFUL!
Please check output for further info!
########### E R R O R ###############
#####################################
The following output is now a tail of the alert.log:
2020-05-20T23:14:57.495022+00:00
JIT: pid 4982 requesting stop
.... (PID:4982): Archival disabled due to shutdown: 1089
Shutting down archive processes
Archiving is disabled
JIT: pid 4982 requesting stop
2020-05-20T23:14:57.551609+00:00
Stopping background process VKTM
2020-05-20T23:15:05.039239+00:00
Instance shutdown complete (OS id: 4982)

```

Reply:
Strange ...